src/coreclr/src/jit/lower.cpp

```diff
@@ -2941,37 +2961,47 @@ void Lowering::LowerRet(GenTreeUnOp* ret)
         BlockRange().InsertBefore(ret, bitcast);
         ContainCheckBitCast(bitcast);
     }
-    else
+    else if (ret->TypeGet() != TYP_VOID)
     {
+        GenTree* retVal = ret->gtGetOp1();
+#if FEATURE_MULTIREG_RET
+        if (op1->OperIs(GT_LCL_VAR) && varTypeIsStruct(op1))
+        {
+            ReturnTypeDesc retTypeDesc;
+            LclVarDsc*     varDsc = nullptr;
+            varDsc                = comp->lvaGetDesc(op1->AsLclVar()->GetLclNum());
+            retTypeDesc.InitializeStructReturnType(comp, varDsc->lvVerTypeInfo.GetClassHandle());
+            if (retTypeDesc.GetReturnRegCount() > 1)
+            {
+                CheckMultiRegLclVar(op1->AsLclVar(), &retTypeDesc);
+            }
+        }
+#else // !FEATURE_MULTIREG_RET
```

Comment:
why was this block put under `!FEATURE_MULTIREG_RET`? 
It breaks `compDoOldStructRetyping == false` logic.

Reply:
I've changed it; I'm not sure why I made that change.